File: applications/UserManagement/Authorization.Common/Models/ApplicationClientModel.cs

namespace Authorization.Common.Models;
public class ApplicationClientModel
{
    /// <summary>
    /// Id of the Client. This will be same as the Object Id of the App Registration
    /// </summary>
    public Guid Id { get; set; }

    /// <summary>
    /// Display name of the Client.
    /// </summary>
    public string Name { get; set; } = default!;

    /// <summary>
    /// Description of the client.
    /// </summary>
    public string Description { get; set; } = default!;

    /// <summary>
    /// Application (Client) Id from the App Registration
    /// </summary>
    public Guid ClientId { get; set; }

    /// <summary>
    /// Willow Application
    /// </summary>
    public ApplicationModel Application { get; set; } = default!;
}

File: applications/UserManagement/Authorization.TwinPlatform.Web/authorization-client-app/src/types/ApplicationClientModel.ts
import * as Yup from 'yup';
import { Guid } from "./Guid";
import { ApplicationModel } from './ApplicationModel';

export class ApplicationClientModel {
  id: string = Guid.Empty.ToString();
  name: string = '';
  description: string = '';
  clientId: string = Guid.Empty.ToString();
  application: ApplicationModel = new ApplicationModel();

  static MapModel(apiRow: any): ApplicationClientModel {
    let newRow = new ApplicationClientModel();
    Object.assign(newRow, apiRow);
    return newRow;
  }

  static validationSchema = Yup.object().shape({
    name: Yup.string().required('Client name is required.').max(100, 'Name cannot be greater than 100 characters in length.'),
    description: Yup.string().required('Description is required.').max(200, 'Description cannot be greater than 500 characters in length.'),
  });

}
